crates/bevy_ecs/src/system/commands/config.rs

```diff
@@ -98,7 +113,7 @@ macro_rules! impl_fallible_commands {
             T: AddCommand,
         {
             /// If the command failed, run the provided `error_handler`.
-            pub fn on_err<'this, H>(&'this mut self, error_handler: H) -> $returnty
+            pub fn on_err_custom<H>(&mut self, error_handler: H) -> $returnty
```

Comment:
Could you add a `BLOCKED:` comment or something similarly appropriate here explaining that we can't condense this into `on_err` because rustc's type inference is too weak?

This would be a nice thing to revisit in a year or three.